test_runner/regress/test_tenant_detach.py

```diff
@@ -167,3 +167,196 @@ def test_detach_while_attaching(
+    tenant_timeline_dir = env.repo_dir / "tenants" / str(tenant_id) / "timelines" / str(timeline_id)
+    layers_removed = False
+    for dir_entry in tenant_timeline_dir.iterdir():
+        if dir_entry.name.startswith("00000"):
+            # Looks like a layer file. Remove it
+            dir_entry.unlink()
+            layers_removed = True
+    assert layers_removed, f"Found no layers for tenant {tenant_timeline_dir}"
+
+    # now, load it from the local files and expect it to work due to remote storage restoration
+    pageserver_http.schedule_tenant_load(tenant_id=tenant_id)
+    wait_until_tenant_status(pageserver_http, tenant_id, "Active", 5)
+
+    tenants_after_attach = [tenant["id"] for tenant in pageserver_http.tenant_list()]
+    tenants_after_attach.sort()
+    assert tenants_after_attach == tenants_before_ignore, "Should have all tenants back"
+
+    timelines_after_ignore = [
+        timeline["timeline_id"] for timeline in pageserver_http.timeline_list(tenant_id=tenant_id)
+    ]
+    assert timelines_before_ignore == timelines_after_ignore, "Should have all timelines back"
+
+    pg.stop()
+    pg.start()
+    with pg.cursor() as cur:
+        assert (
+            query_scalar(cur, f"SELECT secret FROM test WHERE id = {data_id};") == data_secret
+        ), "Should have timeline data back"
+
+
+@pytest.mark.parametrize("remote_storage_kind", [RemoteStorageKind.LOCAL_FS])
+def test_ignored_tenant_stays_broken_without_metadata(
+    neon_env_builder: NeonEnvBuilder, remote_storage_kind: RemoteStorageKind
+):
+    neon_env_builder.enable_remote_storage(
+        remote_storage_kind=remote_storage_kind,
+        test_name="test_ignored_tenant_stays_broken_without_metadata",
+    )
+    env = neon_env_builder.init_start()
+    pageserver_http = env.pageserver.http_client()
+    pg = env.postgres.create_start("main")
+
+    tenant_id = TenantId(pg.safe_psql("show neon.tenant_id")[0][0])
+    timeline_id = TimelineId(pg.safe_psql("show neon.timeline_id")[0][0])
+
+    # ignore the tenant and remove its metadata
+    pageserver_http.tenant_ignore(tenant_id)
+    tenant_timeline_dir = env.repo_dir / "tenants" / str(tenant_id) / "timelines" / str(timeline_id)
+    metadata_removed = False
+    for dir_entry in tenant_timeline_dir.iterdir():
+        if dir_entry.name == "metadata":
+            # Looks like a layer file. Remove it
+            dir_entry.unlink()
+            metadata_removed = True
+    assert metadata_removed, f"Failed to find metadata file in {tenant_timeline_dir}"
+
+    env.pageserver.allowed_errors.append(".*could not load tenant .*?: failed to load metadata.*")
+
+    # now, load it from the local files and expect it to be broken due to inability to load tenant files into memory
+    pageserver_http.schedule_tenant_load(tenant_id=tenant_id)
+    wait_until_tenant_status(pageserver_http, tenant_id, "Broken", 5)
+
+
+# Does not use `wait_until` for debugging purposes
+def wait_until_tenant_status(
```

Comment:
This seems generally useful. Do we have a shared space for such functions?

Reply:
We have a generic https://github.com/neondatabase/neon/blob/main/test_runner/fixtures/utils.py can move it there, but not sure if needed, since this is the only file we need such things so far?

There's `wait_until` from that utils module, but it somehow does not evaluate `log.error` and similar calls inside it and generally is displayed bad in our tests.